pkg/cmd/server/crypto/crypto.go

```diff
@@ -335,10 +337,10 @@ func MakeCA(certFile, keyFile, serialFile, name string) (*CA, error) {
 	}, nil
 }
 
-func (ca *CA) EnsureServerCert(certFile, keyFile string, hostnames sets.String) (*TLSCertificateConfig, bool, error) {
+func (ca *CA) EnsureServerCert(certFile, keyFile string, hostnames sets.String, expirationDate time.Time) (*TLSCertificateConfig, bool, error) {
```

Comment:
I modified many functions here and added one more parameter to them: I'm not sure whether it's ok or not.